tests/bluetooth/tester/src/btp_mcp.c

```diff
@@ -1317,3 +1326,242 @@ uint8_t tester_unregister_mcp(void)
 {
 	return BTP_STATUS_SUCCESS;
 }
+
+/* Media Control Service */
+static uint8_t mcs_supported_commands(const void *cmd, uint16_t cmd_len, void *rsp,
+				      uint16_t *rsp_len)
+{
+	struct btp_mcs_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_MCS_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_MCS_CMD_SEND);
+	tester_set_bit(rp->data, BTP_MCS_CURRENT_TRACK_OBJ_ID_GET);
+	tester_set_bit(rp->data, BTP_MCS_NEXT_TRACK_OBJ_ID_GET);
+	tester_set_bit(rp->data, BTP_MCS_INACTIVE_STATE_SET);
+	tester_set_bit(rp->data, BTP_MCS_PARENT_GROUP_SET);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t mcs_cmd_send(const void *cmd, uint16_t cmd_len, void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_mcs_send_cmd *cp = cmd;
+	struct mpl_cmd mcp_cmd;
+	int err;
+
+	LOG_DBG("MCS Send Command");
+
+	mcp_cmd.opcode = cp->opcode;
+	mcp_cmd.use_param = cp->use_param;
+	mcp_cmd.param = (cp->use_param != 0) ? sys_le32_to_cpu(cp->param) : 0;
+
+	err = media_proxy_ctrl_send_command(mcs_media_player, &mcp_cmd);
+	if (err) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t mcs_next_track_obj_id_get(const void *cmd, uint16_t cmd_len, void *rsp,
+					 uint16_t *rsp_len)
+{
+	struct btp_mcs_next_track_obj_id_rp *rp = rsp;
+	int err;
+
+	LOG_DBG("MCS Read Next Track Obj Id");
+
+	err = media_proxy_ctrl_get_next_track_id(mcs_media_player);
+	if (err) {
+		return BTP_STATUS_FAILED;
+	}
+
+	sys_put_le48(next_track_obj_id, rp->id);
+
+	*rsp_len = sizeof(*rp) + 1;
```

Comment:
this should be sizeof(*rp), shouldn't it?   (there is no variable part here)

Reply:
Fixed